dev-util/pkgconf/pkgconf-1.3.90.recipe

```diff
@@ -0,0 +1,105 @@
+SUMMARY="A lightweight pkg-config replacement"
+DESCRIPTION="A program which helps to configure compiler and linker flags for \
+development libraries. It is similar to pkg-config from freedesktop.org."
+HOMEPAGE="http://pkgconf.org/"
+COPYRIGHT="2011, 2012, 2013, 2014, 2015, 2016, 2017 pkgconf authors"
+LICENSE="ISC"
+REVISION="1"
+SOURCE_URI="https://distfiles.dereferenced.org/pkgconf/pkgconf-$portVersion.tar.xz"
+CHECKSUM_SHA256="112db7f957816901b5fe448715c0a9e38f171d4794c44075b0734a0e7fcfba36"
+PATCHES="pkgconf-$portVersion.patchset"
+
+ARCHITECTURES="!x86_gcc2 x86_64"
+SECONDARY_ARCHITECTURES="x86"
+
+libVersion="2.0.0"
+libVersionCompat="$libVersion compat >= ${libVersion%%.*}"
+
+PROVIDES="
+	pkgconf$secondaryArchSuffix = $portVersion
+	cmd:pkgconf$secondaryArchSuffix = $portVersion
+	cmd:pkg_config$secondaryArchSuffix = $portVersion
```

Comment:
```diff
 ARCHITECTURES="!x86_gcc2 x86_64"
 SECONDARY_ARCHITECTURES="x86"
 
+commandSuffix=$secondaryArchSuffix
+commandBinDir=$binDir
+if [ "$targetArchitecture" = x86_gcc2 ]; then
+	commandSuffix=
+	commandBinDir=$prefix/bin
+fi
+
 libVersion="2.0.0"
 libVersionCompat="$libVersion compat >= ${libVersion%%.*}"
 
 PROVIDES="
 	pkgconf$secondaryArchSuffix = $portVersion
-	cmd:pkgconf$secondaryArchSuffix = $portVersion
-	cmd:pkg_config$secondaryArchSuffix = $portVersion
+	cmd:pkgconf$commandSuffix = $portVersion
 	lib:libpkgconf$secondaryArchSuffix = $libVersionCompat
 	"
```

Reply:
For `pkgconf` I would want the suffix, as a small reminder that on x86_gcc2, it would only works with secondary arch libs.

Reply:
If I'm not mistaken, pkgconfig should be usable on any recipe, including recipes that build on x86_gcc2 primary arch and that do not provide support for building on x86 secondary arch.

So, if this is true, then we really need to provide `cmd:pkgconf$commandSuffix` instead of `cmd:pkgconf$secondaryArchSuffix`.
Otherwise, on a recipe without support for secondary architectures, we would need to hardcode that, when building on x86_gcc2 primary arch, we would require pkgconf_x86.

So, by using `commandSuffix` for `cmd:pkgconf`, we keep things simple, since all cases are handled by:
* a `cmd:pkgconf` (or `cmd:pkg_config`) on recipes that do not define `SECONDARY_ARCHITECTURES`,
* a `cmd:pkgconf` (or `cmd:pkg_config$secondaryArchSuffix`) on recipes that define `SECONDARY_ARCHITECTURES` but do not define a `commandSuffix`,
* a `cmd:pkgconf$commandSuffix` (or `cmd:pkg_config$secondaryArchSuffix`) on recipes that define both `SECONDARY_ARCHITECTURES` and `commandSuffix`.

Reply:
It's basically the same reason why we need two version of `cmd:pkg_config` on x86 secondary arch.

The [`.pc` search path][0] is configured to look into `lib/$secondaryArchSuffix`. So using it for gcc2 wouldn't make any sense, as libs in `lib/x86` are incompatible with gcc2

[0]: https://github.com/haikuports/haikuports/pull/2026/files#diff-d10935f4784240a8d4b63c74253d0880R67

Reply:
BTW, on x86_gcc2 **_non_** hybrid, it won't be possible to use `pkgconf` instead of `pkg-config`, so this means that, as long as x86_gcc2 lives:
1) we will need to keep the `cmd:pkg_config` in `BUILD_PREREQUIRES` for all recipes that build on x86_gcc2 primary arch.
2) it will **_only_** be OK to replace `cmd:pkg_config$secondaryArchSuffix` by `cmd:pkgconf` in those recipes which do **_not_** build on x86_gcc2 primary arch.

Reply:
> It's basically the same reason why we need two version of cmd:pkg_config on x86 secondary arch.

Ok, I see. I guess you are correct.

Reply:
OK, this means pkgconfig won't be usable for building anything on x86_gcc2 primary arch.

So you are perfectly right. The `cmd:pkgconf$secondaryArchSuffix` is correct.

Reply:
Do you think it woud be possible, when building pkgconf on x86 second arch, to change the default search path to that of the primary arch (i.e. `/boot/system/develop/lib/pkgconfig/`), and create a wrapper script, `pkgconf-x86`, that would use the search path for the secondary arch, `/boot/system/develop/lib/x86/pkgconfig/` ?

This would allow, on x86_gcc2 hybrid, to uninstall pkgconfig and install pkgconf_pkgconfig (providing the symlink) and start testing the transition from pkgconfig to pkgconf even before x86_gcc2 is abandonned.

Well, after all, this seems a bit too complicated for a rather small result.
So let's ignore this comment.